### PR TITLE
Vasek/beam rt rebase

### PR DESCRIPTION
**Pull request by VaclavPlajt**
Beam-runtime rebased on master. Please review.

### CODE REVIEW

**Comment by je-ik** [comment]
Please rebase it once more, I have just pushed a few commits je-ik/beam-runtime

**Comment by je-ik** [comment]
Don't worry, I will open PR into this branch

File: euphoria-beam/src/main/java/cz/seznam/euphoria/beam/io/ListCollector.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import cz.seznam.euphoria.core.client.io.Collector;
+import cz.seznam.euphoria.core.client.io.Context;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A context adding values to list.
+ */
+@SuppressWarnings("unchecked")
+public class ListCollector<T> implements Collector<T>, Context {
```
Comment (mareksimunek) [inline euphoria-beam/src/main/java/cz/seznam/euphoria/beam/io/ListCollector.java]: Is it needed? Delete if not.

File: euphoria-beam/src/main/java/cz/seznam/euphoria/beam/BeamExecutor.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    this.accumulatorFactory = accumulatorFactory;
+  }
+
+  /**
+   * Specify global allowed lateness for the executor.
+   * @param duration the allowed lateness for all windows
+   * @return this
+   */
+  public BeamExecutor withAllowedLateness(java.time.Duration duration) {
+    this.allowedLateness = Duration.millis(duration.toMillis());
```
[the remark this answers is not in the document]
Comment (je-ik) [reply euphoria-beam/src/main/java/cz/seznam/euphoria/beam/BeamExecutor.java]: That is problem with current Euphoria's model as a whole. We currently support only single lateness on executor level. Setting this on particular operator requires changes to client API. We probably want to do this in next iteration.

File: euphoria-beam/src/main/java/cz/seznam/euphoria/beam/CombinableCollector.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import cz.seznam.euphoria.core.client.accumulators.Timer;
+import cz.seznam.euphoria.core.client.dataset.windowing.Window;
+import cz.seznam.euphoria.core.client.io.Collector;
+import cz.seznam.euphoria.core.client.io.Context;
+import java.io.Serializable;
+
+/**
+ * {@code Collector} for combinable functors.
+ */
+public class CombinableCollector<T> implements Collector<T>, Serializable {
```
[the remark this answers is not in the document]
Comment (je-ik) [reply euphoria-beam/src/main/java/cz/seznam/euphoria/beam/CombinableCollector.java]: :+1:

**Pull request merged**